ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/JoiningNode.java

```diff
@@ -40,11 +45,30 @@ public interface JoiningNode {
    */
   void setKeyFormat(KeyFormat format);
 
+  static ImmutableMap<String, String> getProperties(FormatInfo formatInfo) {
+    final ImmutableMap.Builder<String, String> propertiesBuilder = ImmutableMap.builder();
+
+    return propertiesBuilder.putAll(
+        formatInfo
+            .getProperties()
+            .entrySet()
+            .stream()
+            .filter(x -> !x.getKey().equals(ConnectProperties.SCHEMA_ID))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+    ).build();
+  }
+
   static ValueFormat getValueFormatForSource(final PlanNode sourceNode) {
```

Comment:
This change is interesting. I think what's happening is that this method name is confusing -- it makes it sound like it returns value formats for source streams/tables (which it does today) but its usage is for internal topics used as part of joins (which is why it's correct to update the format away from what the source format is, to what we want to use for internal topics instead). Is that your understanding of this method as well?

If so, I think I cleaner way to implement this change is to update `InternalFormats.of()` to strip out the schema ID format property (if present) from both key and value formats. If my understanding of the changes in this PR is correct, that should have the same effect and means we can make the changes in this PR in just one place instead of two (JoiningNode for values, and PreJoinRepartitionNode for keys).

A consequence of my suggestion, though, is that other types of operations which require internal topics (group by, aggregate, etc) will also pick up the change to remove schema ID, and not just joins. I think that's a good thing but it'd be good to check -- do you know if group bys and aggregates are affected by the same bug as the join bug that you're trying to resolve? I can't think of a reason that they wouldn't be, but I probably haven't given it as much thought as you have.

Reply:
Yes, I was looking to update the format only for internal topics and found this method being used during joins.

I agree updating the `InternalFormats.of` is a cleaner way to do it.

Looking at the code, it looks like groupbys and aggregates are affected by the same bug since they are also using the same `InternalFormats.of` method. Will add a test for groupby.